google/auth/_credentials_base.py

```diff
@@ -62,7 +63,7 @@ def refresh(self, request):
         # (pylint doesn't recognize that this is abstract)
         raise NotImplementedError("Refresh must be implemented")
 
-    def _apply(self, headers, token=None):
+    def _apply(self, headers: dict[str, str], token: Optional[str] = None):
```

Comment:
Shall we use `Mapping` from typing (here and in other places)? Also update the docstring to `Mapping[str, str]`.

Reply:
This operation requires the type of `headers` has `__setitem__` method that's why being a `Mapping` is not enough:

```py
headers["authorization"] = "Bearer {}".format(
    _helpers.from_bytes(token or self.token)
)
```

Reply:
Do you want me to modify the doctring instead?